src/analyzer/analyzerbeats.cpp

```diff
@@ -38,7 +39,7 @@ bool AnalyzerBeats::initialize(TrackPointer tio, int sampleRate, int totalSample
         return false;
     }
 
-    bool bPreferencesBeatDetectionEnabled = static_cast<bool>(
+    bool bPreferencesBeatDetectionEnabled = m_batch || static_cast<bool>(
```

Comment:
Ok than I think m_batch should become m_forceBeatDetection and bPreferencesBeatDetectionEnabled should become beatDetectionEnabled.